### PR TITLE
Yaml loader for recipes

### PR DESCRIPTION
**Pull request by arnaudon**

### CODE REVIEW

**Comment by arnaudon** [comment]
Ah sorry, only the last commits should be in the PR, the others are for another PR. I'll remove them.

File: bluepymm/prepare_combos/parse_files.py [hunk 1]
```diff
@@ -92,6 +92,52 @@ def read_mm_recipe(recipe_filename):
...
+        recipe_filename(str): filename of recipe (XML/YAML)
+
+    Returns:
+        A pandas.DataFrame with fields "layer", "fullmtype", and "etype".
+    """
+    from pathlib import Path
+
+    if Path(recipe_filename).suffix == '.xml':
+        return read_mm_recipe_xml(recipe_filename)
+    if Path(recipe_filename).suffix == '.yaml':
```
Comment (wvangeit) [inline bluepymm/prepare_combos/parse_files.py:105]: This should ideally be, 
if...
elif..
else..

File: bluepymm/prepare_combos/parse_files.py [hunk 2]
```diff
@@ -92,6 +92,52 @@ def read_mm_recipe(recipe_filename):
...
+    with open(recipe_filename, 'r') as f:
+        recipe = yaml.safe_load(f)
+
+    assert recipe['version'] in ('v2.0',)
+
+    mecombos = pandas.DataFrame(columns=["layer", "fullmtype", "etype"])
+    for region in recipe['neurons']:
+        for etype in region['traits']['etype'].keys():
+            n_combos = len(mecombos)
+            mecombos.loc[n_combos, 'layer'] = region['traits']['layer']
```
Comment (wvangeit) [inline bluepymm/prepare_combos/parse_files.py:131]: there is no n-1 necessary here?

Comment (arnaudon) [reply bluepymm/prepare_combos/parse_files.py:132]: yes, I'm stupid

File: bluepymm/prepare_combos/parse_files.py [hunk 3]
```diff
@@ -92,6 +92,53 @@ def read_mm_recipe(recipe_filename):
...
+
+    Returns:
+        A pandas.DataFrame with fields "layer", "fullmtype", and "etype".
+    """
+    import yaml
+
+    with open(recipe_filename, 'r') as f:
+        recipe = yaml.safe_load(f)
+
+    assert recipe['version'] in ('v2.0',)
```
Comment (anilbey) [inline bluepymm/prepare_combos/parse_files.py:126]: maybe an exception can be raised here instead of the assertion

Comment (wvangeit) [reply bluepymm/prepare_combos/parse_files.py:126]: I agree

File: bluepymm/prepare_combos/parse_files.py [hunk 4]
```diff
@@ -92,6 +92,53 @@ def read_mm_recipe(recipe_filename):
     """Read a BBP builder recipe and return a pandas.DataFrame with all
     possible (layer, m-type, e-type)-combinations.
 
+    Args:
+        recipe_filename(str): filename of recipe (XML/YAML)
+
+    Returns:
+        A pandas.DataFrame with fields "layer", "fullmtype", and "etype".
+    """
+    from pathlib import Path
```
Comment (wvangeit) [inline bluepymm/prepare_combos/parse_files.py:101]: this seems not python 2 compatible

Comment (wvangeit) [reply bluepymm/prepare_combos/parse_files.py]: one should use os.path.splitext

**Comment by arnaudon** [comment]
do you want the test?

**Comment by anilbey** [comment]
I am working on it

**Comment by wvangeit** [comment]
I'll already merge, but yeah, @anilbey please also add a test

**Comment by anilbey** [comment]
Ok but I cannot push it any longer let me create a separate pr

**Comment by anilbey** [comment]
Btw. there are two issues here. 
* Pyyaml import is missing - it's added in #199 
* Yaml parser is parsing everything as the object type

File: bluepymm/prepare_combos/parse_files.py (hunk 3 above)
Comment (mgeplf) [reply bluepymm/prepare_combos/parse_files.py]: I use assertions for code coverage reasons - for full coverage, I'd need to make a fake yaml file w/ the wrong version, which seems excessive.
The assert still raises an exception, and the message can be made informative.  I doubt anyone will try and catch this exception, so whether it's an `Exception` or `AssertionError`, I'm not sure it matters.

Comment (anilbey) [reply bluepymm/prepare_combos/parse_files.py]: I see your point. But the assert statement is removed when Python is called with the -O flag. I don't know if a python rpm or deb package is using the -O flag.

Comment (mgeplf) [reply bluepymm/prepare_combos/parse_files.py]: You have to explictly enable 'optimizations' (ie: call `python -O ....`), and the performance difference b/c of that is pretty small - I've actually never seen anyone do that.  Hence why I'm not generally worried about using asserts.

Comment (anilbey) [reply bluepymm/prepare_combos/parse_files.py]: I see. Would a one liner `if not condition: raise Exception()` not be the same in terms of coverage?

Comment (mgeplf) [reply bluepymm/prepare_combos/parse_files.py]: not for branch coverage: you'd have to explicitly write tests where `condition` is true, and false

Comment (anilbey) [reply bluepymm/prepare_combos/parse_files.py]: Let's create a PEP with the `raiseif` statement :))

**Pull request merged**